### ISSUE TITLE
Certain Unary expressions in match statements break parser

### ISSUE THREAD

**Issue by tonybaloney** [microsoft/pyright#5798]
## Environment data

-   Language Server version: Pylance language server 2023.8.30 (pyright 634c8074)
-   OS and version: macOS 13.5
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.12 a4 

## Code Snippet

A match statement takes an expression as its first node. I've just noticed that Pylance is reporting syntax errors when the positive unary operator is used with a parenthesised expression and a match statement. 

Take this minimal example:

```python
match +(1 - 5):
    case _:
        print("yikes")

```

In Python, this code prints "yikes", but in VS Code I get an error on every line:

<img width="614" alt="screenshot 2023-08-24 at 09 47 13" src="https://github.com/microsoft/pylance-release/assets/1532417/1af852d5-3a19-411c-a781-a62f24b96705">

When you replace it with a unary negative operator, the errors go away (but both are valid Python)

<img width="352" alt="screenshot 2023-08-24 at 09 47 47" src="https://github.com/microsoft/pylance-release/assets/1532417/d4c35d40-2d80-4129-8aef-981372677032">

## Logs

```
XXX
```

**Comment by erictraut** [comment]
This is included in pyright 1.1.325, which I just published. It will also be included in an upcoming pylance release.
